pytorch_lightning/utilities/auto_restart.py

```diff
@@ -656,6 +646,8 @@ def __accumulate_state(self, sampler_state: Dict[str, Any]) -> None:
 
     def _store_sampler_state(self) -> None:
         """This function is used to extract the sampler states if any."""
+        from pytorch_lightning.trainer.connectors.checkpoint_connector import _SupportsStateDict
+
```

Comment:
nit: Shouldn't this import statement be at the top?

Reply:
was causing some issue while running tests.. not sure why.

Reply:
must be a circular import, looks ok to me. could add a comment

Reply:
Or move it to `types.py`. It would solve the circular dependency

Reply:
done!